src/dockerhub.ts

```diff
@@ -15,113 +15,202 @@
+    validateRepoParts(req);
+
+    req.full_description = req.full_description ?? '';
+
+    if (req.full_description.trim() === '' && !req.allow_empty_full_description) {
+      throw new Error(`DockerApiClient.updateRepositoryDescription - req.full_description is empty and req.allow_empty_full_description is false`);
+    }
 
-  public async updateRepoDescription(req: UpdateRepoDescriptionRequest): Promise<RepositoryResponse> {
-    const body = {
+    const body: Record<string, string> = {
       full_description: req.full_description
     };
-    if (req.description) {
-      body['description'] = req.description;
+
+    if ((req.description ?? '').trim() != '') {
+      body['description'] = req.description as string;
+    }
+
+    try {
+      const rawResp = await this.client.patch(`${apiBaseUrl}/repositories/${req.namespace}/${req.name}`, JSON.stringify(body));
+
+      return handleResponse<RepositoryResponse>(rawResp);
+    } catch (e) {
+      throw new Error(`DockerApiClient.updateRepositoryTags - ${e}`);
     }
-    const resp: httpm.HttpClientResponse = await this.httpc.patch(`${repositoriesURL}${req.namespace}/${req.name}`, JSON.stringify(body));
-    return <RepositoryResponse>JSON.parse(await DockerHub.handleResponse(resp));
   }
+}
 
-  private static async login(req: TokenRequest): Promise<string> {
-    const http: httpm.HttpClient = new httpm.HttpClient('docker-actions-toolkit', [], {
-      headers: {
-        'Content-Type': 'application/json'
-      }
-    });
-    const resp: httpm.HttpClientResponse = await http.post(loginURL, JSON.stringify(req));
-    const tokenResp = <TokenResponse>JSON.parse(await DockerHub.handleResponse(resp));
-    core.setSecret(`${tokenResp.token}`);
-    return `${tokenResp.token}`;
+export interface RepoParts {
+  namespace: string;
+  name: string;
+}
+
+export function validateRepoParts(val: RepoParts) {
+  val = val ?? {
+    namespace: '',
+    name: ''
+  };
+  val.namespace = val.namespace ?? '';
+  val.name = val.name ?? '';
+
+  if (val.namespace.trim() == '') {
+    throw new Error('req.namespace is required');
   }
 
-  private static async handleResponse(resp: httpm.HttpClientResponse): Promise<string> {
-    const body = await resp.readBody();
-    resp.message.statusCode = resp.message.statusCode || HttpCodes.InternalServerError;
-    if (resp.message.statusCode < 200 || resp.message.statusCode >= 300) {
-      if (resp.message.statusCode == HttpCodes.Unauthorized) {
-        throw new Error(`Docker Hub API: operation not permitted`);
-      }
-      const errResp = <Record<string, string>>JSON.parse(body);
-      for (const k of ['message', 'detail', 'error']) {
-        if (errResp[k]) {
-          throw new Error(`Docker Hub API: bad status code ${resp.message.statusCode}: ${errResp[k]}`);
-        }
+  if (val.name.trim() == '') {
+    throw new Error('req.name is required');
+  }
+}
```

Comment:
Is it really necessary? I would prefer this to be handled server side and I think this is already the case no?